res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,978 @@
+
+    // Hotcues mode is default start value
+    TraktorS2MK3.controller.setOutput("[Channel1]", "hotcues", valueFullLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "hotcues", valueFullLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "samples", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "samples", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "keylock", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "keylock", valueSoftLight, false);
+
+    for (var i = 1; i <= 8; ++i) {
+        TraktorS2MK3.controller.setOutput("[Channel1]", "hotcue_" + i + "_enabled", valueSoftLight, false);
+        TraktorS2MK3.controller.setOutput("[Channel2]", "hotcue_" + i + "_enabled", valueSoftLight, false);
+    }
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "pfl", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "pfl", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[ChannelX]", "fxButton1", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[ChannelX]", "fxButton2", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[ChannelX]", "fxButton3", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[ChannelX]", "fxButton4", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "reverse", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "reverse", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "flx", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "flx", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "addTrack", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "addTrack", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "grid", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "grid", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[Channel1]", "MaximizeLibrary", valueSoftLight, false);
+    TraktorS2MK3.controller.setOutput("[Channel2]", "MaximizeLibrary", valueSoftLight, false);
+
+    TraktorS2MK3.controller.setOutput("[ChannelX]", "quantize", valueSoftLight, false);
+
+    // For the last output we should send the packet finally
+    TraktorS2MK3.controller.setOutput("[Microphone]", "talkover", valueSoftLight, true);
+};
+
+TraktorS2MK3.messageCallback = function (packet, data) {
+    for (var name in data) {
+        if (data.hasOwnProperty(name)) {
+            TraktorS2MK3.controller.processButton(data[name]);
+        }
+    }
+};
+
+TraktorS2MK3.shutdown = function () {
+    // Disconnect VuMeter callbacks
+    this.vuLeftConnection.disconnect();
+    this.vuRightConnection.disconnect();
+    this.clipLeftConnection.disconnect();
+    this.clipRightConnection.disconnect();
```

Comment:
You don't need to do this in the script. Mixxx deletes all callback connections on shutdown.

Reply:
Removed them in https://github.com/mi01/mixxx/commit/3f0fd915ae129993968be79bcb867eb8f22bb468